sqlite/ext/comdb2/limits.c

```diff
@@ -0,0 +1,203 @@
+/*
+   Copyright 2017 Bloomberg Finance L.P.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+#if (!defined(SQLITE_CORE) || defined(SQLITE_BUILDING_FOR_COMDB2)) &&          \
+    !defined(SQLITE_OMIT_VIRTUALTABLE)
+
+#if defined(SQLITE_BUILDING_FOR_COMDB2) && !defined(SQLITE_CORE)
+#define SQLITE_CORE 1
+#endif
+
+#include <assert.h>
+#include "comdb2systbl.h"
+#include "comdb2systblInt.h"
+#include "cdb2_constants.h"
+
+#define COMDB2_LIMIT(LIMIT, DESCR)                                             \
+    {                                                                          \
+        #LIMIT, DESCR, LIMIT                                                   \
+    }
+
+struct limit_t {
+    const char *name;
+    const char *descr;
+    int limit;
+} limits[] = {
+    /* COMDB2_LIMIT(COMDB2_MAX_RECORD_SIZE, "????"), */
+    /* COMDB2_LIMIT(LONG_REQMS, "????"), */
+    COMDB2_LIMIT(MAXBLOBLENGTH, "Maximum blob length"),
+    COMDB2_LIMIT(MAXBLOBS, "Maximum number of blobs"),
+    COMDB2_LIMIT(MAXCOLNAME, "Maximum column name length"),
+    COMDB2_LIMIT(MAXCOLUMNS, "Maximum columns in a table"),
+    COMDB2_LIMIT(MAXCONSTRAINTS, "Maximum number of constraint in a table"),
+    COMDB2_LIMIT(MAXCONSUMERS, "Maximum queue consumers"),
+    COMDB2_LIMIT(MAXCUSTOPNAME, "Maximum length of a custom operation name"),
+    COMDB2_LIMIT(MAX_DBNAME_LENGTH, "Maximum length of database name"),
+    COMDB2_LIMIT(MAXDTASTRIPE, "Maximum number of data stripes"),
+    /* COMDB2_LIMIT(MAXDYNTAGCOLUMNS, "????"), */
```

Comment:
MAXDYNTAGCOLUMNS - effectively the max number of bound parameter values per statement.